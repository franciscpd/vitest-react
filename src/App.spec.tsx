import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import App from "./App";

describe("App Component", () => {
  it("should render component", () => {
    const { getByText, container } = render(<App />);

    expect(getByText("Vite + React")).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render correct counter", async () => {
    const { getByText, findByText } = render(<App />);
    const btnCounter = getByText("count is 0");

    userEvent.click(btnCounter);

    const newValue = await findByText("count is 1");
    expect(newValue).toBeInTheDocument();
  });
});