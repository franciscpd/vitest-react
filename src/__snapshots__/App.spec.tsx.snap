// Vitest Snapshot v1

exports[`App Component > should render component 1`] = `
<div
  class="App"
>
  <div>
    <a
      href="https://vitejs.dev"
      target="_blank"
    >
      <img
        alt="Vite logo"
        class="logo"
        src="/vite.svg"
      />
    </a>
    <a
      href="https://reactjs.org"
      target="_blank"
    >
      <img
        alt="React logo"
        class="logo react"
        src="/src/assets/react.svg"
      />
    </a>
  </div>
  <h1>
    Vite + React
  </h1>
  <div
    class="card"
  >
    <button>
      count is 
      0
    </button>
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to test HMR
    </p>
  </div>
  <p
    class="read-the-docs"
  >
    Click on the Vite and React logos to learn more
  </p>
</div>
`;
